### ISSUE TITLE
Audit log is not output after the SET ROLE.

### ISSUE THREAD

**Issue by harada-toshi** [pgaudit/pgaudit#6]
#  Audit log is not output after the SET ROLE.

Hi.

I am testing the pgaudit(https://commitfest.postgresql.org/9/463/).
(use http://www.postgresql.org/message-id/56B0101B.6070704@pgmasters.net attached patch on 9.6-devel)

I found strange thing.

- After SET ROLE, part of the SQL is not the audit log output.
- SQL comprising a relation is not output to the audit log.

# Reproduce:
## prepare
```
createuser test_user -U postgres
createdb test -U postgres -O test_user
psql test -U test_user -c "CREATE TABLE team(id int, name text)"
```
## pgaudit settings

```
shared_preload_libraries = 'pgaudit'
pgaudit.log = 'all'
```
## test sql script (test.sql)

```
SELECT 1;
SELECT * FROM team; -- output audit log
SET ROLE test_user;
SELECT 2;
SELECT * FROM team; -- no output audit log
SELECT 3;
RESET ROLE;
SELECT * FROM team; -- output audit log
```
## run script

```
psql test -U postgres -f test.sql
```

## audit log

```
LOG:  AUDIT: SESSION,1,1,READ,SELECT,,,SELECT 1;,<not logged>
LOG:  AUDIT: SESSION,2,1,READ,SELECT,,,SELECT * FROM team;,<not logged>
LOG:  AUDIT: SESSION,3,1,MISC,SET,,,SET ROLE test_user;,<not logged>
LOG:  AUDIT: SESSION,4,1,READ,SELECT,,,SELECT 2;,<not logged>
LOG:  AUDIT: SESSION,5,1,READ,SELECT,,,SELECT 3;,<not logged>
LOG:  AUDIT: SESSION,6,1,MISC,RESET,,,RESET ROLE;,<not logged>
LOG:  AUDIT: SESSION,7,1,READ,SELECT,,,SELECT * FROM team;,<not logged>
```

**Comment by dwsteele** [comment]
Hi, thanks for submitting this.  I found the issue and you can see the fix on this branch:

https://github.com/pgaudit/pgaudit/commit/8162d32faf68f243d8d5fbfc627eeaff4d553848

The code was intended to suppress FK lookups and security definer functions to keep the logs from being too noisy.  This fix definitely helps you problem but it will make superuser sessions much noiser.

Take a look and let me know what you think.

**Comment by harada-toshi** [comment]
Hello.

With this fix(8162d32), the problem that I have reported was able to confirm that it has solved.
Thank you very much.

The following describes the my personal thoughts.
* Operation by the super user, should be audited, including the operation that is derived.
* In a typical system, the operation by the super user is done only at the time of maintenance.
* Therefore the superuser's audit log is also no problem by noisiness.
